netbox_cmdb/netbox_cmdb/models/interface.py

```diff
@@ -0,0 +1,172 @@
+        blank=True,
+        null=True,
+        help_text="Interface speed in kB/s",
+    )
+    fec = models.CharField(
+        choices=FEC_CHOICES,
+        max_length=5,
+        blank=True,
+        null=True,
+    )
+    description = models.CharField(max_length=100, blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.device.name}--{self.name}"
+
+    class Meta:
+        unique_together = ("device", "name")
+
+
+class LogicalInterface(ChangeLoggedModel):
+    """A logical interface configuration."""
+
+    index = models.PositiveSmallIntegerField()
+    parent_interface = models.ForeignKey(
+        to="DeviceInterface",
+        related_name="%(class)s",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+    )
+    mtu = models.PositiveIntegerField(blank=True, null=True)
+    type = models.CharField(
+        choices=LOGICAL_INTERFACE_TYPE_CHOICES,
+        max_length=2,
+        default=None,
+    )
+    vrf = models.ForeignKey(
+        to="VRF", related_name="%(class)s_vrf", on_delete=models.CASCADE, blank=True, null=True
+    )
+    ipv4_address = models.ForeignKey(
+        to="ipam.IPAddress",
+        related_name="%(class)s_ipv4_address",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+    )
+    ipv6_address = models.ForeignKey(
+        to="ipam.IPAddress",
+        related_name="%(class)s_ipv6_address",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+    )
+    mode = models.CharField(
+        choices=LOGICAL_INTERFACE_MODE_CHOICES,
+        blank=True,
+        null=True,
+        default=None,
+        max_length=20,
+        help_text="Interface mode (802.1Q)",
+    )
+    untagged_vlan = models.ForeignKey(
+        to="VLAN",
+        related_name="%(class)s_untagged_vlan",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+        default=None,
+    )
+    tagged_vlans = models.ManyToManyField(
+        to="VLAN", related_name="%(class)s_tagged_vlans", blank=True, default=None
+    )
+    native_vlan = models.ForeignKey(
+        to="VLAN",
+        related_name="%(class)s_native_vlan",
+        on_delete=models.CASCADE,
+        blank=True,
+        null=True,
+        default=None,
+    )
+    description = models.CharField(max_length=100, blank=True, null=True)
+
+    def __str__(self):
+        return f"{self.parent_interface.name}--{self.index}"
+
+    class Meta:
+        unique_together = ("index", "parent_interface")
+
+
+class Link(ChangeLoggedModel):
+    """A link between two DeviceInterface."""
+
+    interface_a = models.ForeignKey(
+        to="DeviceInterface",
+        related_name="%(class)s_interface_a",
+        on_delete=models.CASCADE,
+    )
+    interface_b = models.ForeignKey(
+        to="DeviceInterface",
+        related_name="%(class)s_interface_b",
+        on_delete=models.CASCADE,
+    )
+    state = models.CharField(
```

Comment:
As the state is on the link, how do we put a "logical" link in maintenance.
For instance:
* We have a physical link between 2 devices
* There are multiple vlans (trunk link), so several logical interface
* I want to put in maintenance only a single logical interface (by shutting it down), how do I put in in maintenance ?